inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java

```diff
@@ -175,6 +174,15 @@ public TaskResult getTaskResult(TaskRequest request) {
         List<StreamSourceEntity> needIssuedList = sourceMapper.selectByStatusAndIp(statusList, agentIp, uuid);
         entityList.addAll(needIssuedList);
 
+        List<StreamSourceEntity> fileEntityList = sourceMapper.selectByStatusAndType(needAddStatusList,
+                Lists.newArrayList(SourceType.FILE.getType()), TASK_FETCH_SIZE * 2);
+        for (StreamSourceEntity fileEntity : fileEntityList) {
+            FileSourceDTO fileSourceDTO = FileSourceDTO.getFromJson(fileEntity.getExtParams());
+            if (agentIp.equals(fileSourceDTO.getIp())) {
```

Comment:
Why not put the `agentIp` into the `selectByStatusAndType` method to filter result?

Reply:
1 agentIp is empty if not fetched by agent
2 selectByType is useful sooner or later